openshift/template.yaml

```diff
@@ -111,6 +111,11 @@ objects:
             value: ${STACK_ANALYSIS_REQUEST_TIMEOUT}
           - name: FLASK_LOGGING_LEVEL
             value: ${FLASK_LOGGING_LEVEL}
+          - name: APP_SECRET_KEY
+            valueFrom:
+              secretKeyRef:
+                name: jobs
+                key: flask-app-secret-key
```

Comment:
This looks so generic? BTW, why do you need a secrete to access internal service?

Reply:
In initial discussion about Ingestion API it was decided that we will not use any token which can will expire after some time, so we can use a secret for validation.
Yes it is used internally as of now so validation is not necessary, but it is part of Jobs APIs and its route is exposed outside, so it would be to keep some kind of authentication.

Reply:
I would suggest two endpoints, one for internal and another for external(both can call same function). From the Route description you can use `host` property for external api. Just a thought.

@yzainee

Reply:
what advantage would that have? having 2 APIs doing the same stuff?

Reply:
@yzainee No shared secrete hassle involved

Reply:
I am not for or against having the 2 endpoints. Will leave it upto you how you want to go ahead.

Reply:
Yes, I will do that as it was discussed in meetings and everyone was in agreement. I have not done that yet because this is already being used by repos like reports and github refresh, If I change API auth then I need to update it in other repos immediately, that is why I have not done any changes in endpoints as of now, I will create new end points and one by one I will update them in all consumer repos. so it wont block us from starting unknown flow ingestion and github refresh, unknwon version ingestion will also work fine, in the meantime we introduce new end point without authentication.

Reply:
@yzainee @arajkumar if you are Okay with it and if there are no code changes required, can you please approve it?